### PR TITLE
build: update dockerfiles

### PR DESCRIPTION
**Pull request by colinlyguo**
### Purpose or design rationale of this PR

Some images met build failures with recent changes in [da-codec repo](https://github.com/scroll-tech/da-codec), e.g., https://github.com/scroll-tech/scroll/actions/runs/9872236028
Remove unused `COPY ./prover/go.* ./prover/`.

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] fix: A bug fix

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change

### CODE REVIEW

**Comment by colinlyguo** [comment]
will deploy them in sepolia and test them.

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -2,8 +2,6 @@ name: Docker
 
 on:
   push:
-    tags:
-      - v**
```
Comment (Thegaram) [inline .github/workflows/docker.yml]: Why remove these? Do you just want to trigger this workflow manually?

Comment (georgehao) [inline .github/workflows/docker.yml]: can't remove this

**Review by Thegaram — DISMISSED** [review]

File: .github/workflows/docker.yml (hunk 1 above)
Comment (colinlyguo) [reply .github/workflows/docker.yml]: yes. just testing. reverted it back now.

Comment (colinlyguo) [reply .github/workflows/docker.yml]: reverted back, just used for testing.

**Review by georgehao — APPROVED** [review]

File: .github/workflows/docker.yml (hunk 1 above)
Comment (colinlyguo) [reply .github/workflows/docker.yml]: https://github.com/scroll-tech/scroll/actions/runs/9872663964
all passed here.

Comment (colinlyguo) [reply .github/workflows/docker.yml]: all built successfully here: https://github.com/scroll-tech/scroll/actions/runs/9872663964

**Comment by 0xmountaintop** [comment]
just curious, why do we need to bump version?

**Review by 0xmountaintop — APPROVED** [review]

**Comment by colinlyguo** [comment]
> just curious, why do we need to bump version?

wanna deploy with the new version, since v4.4.25 is already used.

**Pull request merged**